ocaml-lsp-server/src/semantic_highlighting.ml

```diff
@@ -88,68 +88,93 @@ module Token_type = struct
     | Decorator -> 22
 end [@warning "-37"]
 
-module Token_modifier = struct
-  type t =
-    | Declaration
-    | Definition
-    | Readonly
-    | Static
-    | Deprecated
-    | Abstract
-    | Async
-    | Modification
-    | Documentation
-    | Default_library
+module Token_modifiers : sig
+  type t
```

Comment:
I would rather organize it like this:

```
module Token_modifier : sig
  type t
  ..
  module Set  : sig
    type elt := set
    type t
    val empty
    val add : t -> elt -> t
  end
end
```

Reply:
What you propose is more intuitive indeed but also more verbose. The current solution we have reminds of a monoid where one can `append` values to each other and we have an identity element `empty`. I can use `<+>` instead of `%%` for more monoid-like notation and maybe can add a comment. Can also rewrite to you version but seems a little verbose

Reply:
I looked at it again and you can leave it as is. I don’t see how one can do something wrong even if they mix up sets and their elements with the current interface. Can you switch %%% to `union` though? Verbose or not, I much prefer using words over operators.